Realm/RLMObjectStore.mm

```diff
@@ -170,6 +171,26 @@ void RLMAddObjectToRealm(__unsafe_unretained RLMObjectBase *const object,
     return object;
 }
 
+RLMObjectBase *RLMObjectFromObjLink(RLMRealm *realm, realm::ObjLink&& objLink, bool parentIsSwiftObject) {
+    if (auto tableInfo = realm->_info[objLink.get_table_key()]) {
+        return RLMCreateObjectAccessor(*tableInfo, objLink.get_obj_key().value);
+    } else {
+        // Construct the object dynamically.
+        // This code path should only be hit on first access of the object.
+        Class cls = parentIsSwiftObject ? [RealmSwiftObject class] : [RLMDynamicObject class];
+        realm::Schema const groupSchema = realm::ObjectStore::schema_from_group(realm->_realm->read_group());
+        realm::ObjectSchema *schema = new realm::ObjectSchema;
```

Comment:
```suggestion
        auto& group = realm->_realm->read_group();
        auto schema = std::make_unique<realm::ObjectSchema>(group, group.get_table_name(objLink.get_table_key()), objLink.get_table_key());
```

No need to construct the entire Schema just to get a single ObjectSchema, and the object should be passed around as a unique_ptr.